modin/experimental/core/execution/native/implementations/omnisci_on_native/expr.py

```diff
@@ -219,6 +219,11 @@ def cast(self, res_type):
         BaseExpr
             The cast expression.
         """
+        # From float to int cast we expect truncate behavior but CAST
+        # operation would give us round behavior.
+        if is_float_dtype(self._dtype) and is_integer_dtype(res_type):
+            return self.floor()
```

Comment:
Is this just `floor` the column without casting the floored result to int? `FLOOR` in OmniSci seems to not cast it to int but just discards the decimal part, if we want to get an integer in the result we have to explicitly cast it
```python
(Pdb) server.select_df("SELECT * FROM test_table")
     a
0  1.2
1  1.5
2  1.6
(Pdb) server.select_df("SELECT FLOOR(a) FROM test_table")
   EXPR$0
0     1.0
1     1.0
2     1.0
(Pdb) server.select_df("SELECT CAST(FLOOR(a) as INT) FROM test_table")
   EXPR$0
0       1
1       1
2       1
```

Reply:
Nice catch. In our case, the `floor` expression produces an integer result. But it's possible that we do actual cast on the Modin side. I will check it and make a new patch if required since it's not related to this issue.

Reply:
Checked the actual request and result from OmniSci for `FLOOR` expression. Here is what we send:
```
"exprs": [{"op": "FLOOR", "operands": [{"input": 0}], "type": {"type": "BIGINT", "nullable": true}}]
```
Looks like using `BIGINT` in op type is enough to get the actual cast to int. The resulting Arrow table holds integer values.